src/context/wishlist-context.js

```diff
@@ -0,0 +1,67 @@
+import axios from "axios";
+import { useState, useEffect, createContext, useContext } from "react";
+
+const WishlistContext = createContext();
+
+const WishlistProvider = ({ children }) => {
+	const [wishlistProducts, setWishlistProducts] = useState([]);
+	const encodedToken =
+		"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI4NDA1MjA4MS1hNjAwLTQ2YmQtYTNhZS0yZDljNTU2YTQ0NzgiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.gaqvCVkY0hv54te82TnyI8W3iLXNbRScUytBNaTE3GM";
+
+	const addToWishlist = async (product) => {
+		try {
+			const response = await axios.post(
+				"/api/user/wishlist",
+				{ product },
+				{ headers: { authorization: encodedToken } }
+			);
+			setWishlistProducts(response.data.wishlist);
+			console.log(response);
+		} catch (error) {
+			console.error(error);
+		}
+	};
+
+	const removeFromWishlist = async (productId) => {
+		try {
+			const response = await axios.delete(
+				`/api/user/wishlist/${productId}`,
+				// { product },
+				{ headers: { authorization: encodedToken } }
```

Comment:
you can keep the header global so that you dont have to add them in each function call